src/sql/workbench/parts/dataExplorer/common/nodeContextUtils.ts

```diff
@@ -0,0 +1,159 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as os from 'os';
+import { INodeContextValue } from 'sql/workbench/parts/dataExplorer/common/nodeContext';
+import { RawContextKey, IContextKeyService, IContextKey } from 'vs/platform/contextkey/common/contextkey';
+import { Disposable } from 'vs/base/common/lifecycle';
+import { IConnectionManagementService } from 'sql/platform/connection/common/connectionManagement';
+import { ConnectionProfile } from 'sql/platform/connection/common/connectionProfile';
+import { ICapabilitiesService } from 'sql/platform/capabilities/common/capabilitiesService';
+import { mssqlProviderName } from 'sql/platform/connection/common/constants';
+import { NodeType } from 'sql/workbench/parts/objectExplorer/common/nodeType';
+import { ExtensionNodeType } from 'sql/workbench/api/common/sqlExtHostTypes';
+
+export class NodeContextUtils extends Disposable {
+
+	static readonly canSelect = new Set([NodeType.Table, NodeType.View]);
+	static readonly canEditData = new Set([NodeType.Table]);
+	static readonly canCreateOrDelete = new Set([NodeType.AggregateFunction, NodeType.PartitionFunction, NodeType.ScalarValuedFunction,
+	NodeType.Schema, NodeType.StoredProcedure, NodeType.Table, NodeType.TableValuedFunction,
+	NodeType.User, NodeType.UserDefinedTableType, NodeType.View]);
+	static readonly canExecute = new Set([NodeType.StoredProcedure]);
+	static readonly canAlter = new Set([NodeType.AggregateFunction, NodeType.PartitionFunction, NodeType.ScalarValuedFunction,
+	NodeType.StoredProcedure, NodeType.TableValuedFunction, NodeType.View]);
+
+	// General node context keys
+	static NodeProvider = new RawContextKey<string>('nodeProvider', undefined);
+	static IsDatabaseOrServer = new RawContextKey<boolean>('isDatabaseOrServer', false);
+	static IsWindows = new RawContextKey<boolean>('isWindows', os.platform() === 'win32');
+	static IsCloud = new RawContextKey<boolean>('isCloud', false);
+	static NodeType = new RawContextKey<string>('nodeType', undefined);
+	static NodeLabel = new RawContextKey<string>('nodeLabel', undefined);
+
+	// Scripting context keys
+	static CanScriptAsSelect = new RawContextKey<boolean>('canScriptAsSelect', false);
+	static CanEditData = new RawContextKey<boolean>('canEditData', false);
+	static CanScriptAsCreateOrDelete = new RawContextKey<boolean>('canScriptAsCreateOeDelete', false);
+	static CanScriptAsExecute = new RawContextKey<boolean>('canScriptAsExecute', false);
+	static CanScriptAsAlter = new RawContextKey<boolean>('canScriptAsAlter', false);
```

Comment:
At the very least lets pull the mapping of Node Type <-> Supported Action (Create/Drop/Alter/Edit/etc) into a separate file and then have this and objectExplorerActions both use that mapping to add in the appropriate actions/menu items.

But if we can it would be even better to try and think if we can make this even more robust - for example whether we could use the SMO objects themselves to determine what they support (especially since the requests to do the create/script/drop/etc end up going there anyways)